src/tv2-common/parts/effekt.ts

```diff
@@ -51,15 +52,34 @@ export function CreateEffektForPartBase(
 		)
 
 		return ret ?? {}
-	} else if (transition !== undefined && transition.duration !== undefined) {
-		if (transition.style.match(/mix/i)) {
-			return CreateMixForPartInner(pieces, partDefinition.externalId, transition.duration, layers) ?? {}
-		} else {
-			return {}
-		}
-	} else {
+	}
+
+	if (transition === undefined || transition.duration === undefined) {
 		return {}
 	}
+
+	if (transition.style.match(/mix/i)) {
+		return (
+			CreateMixEffectBlueprintPieceForPart(
+				pieces,
+				partDefinition.externalId,
+				transition.duration,
+				layers.sourceLayer
+			) ?? {}
+		)
+	}
+	if (transition.style.match(/dip/i)) {
```

Comment:
A lot of these regexes. I was hoping you'll refactor `PartTransition` so that `style` is an enum.  (and some other places that lead to us using regexes in `getExistingTransition` too)

Reply:
That thought didn't even occur to me. Refactoring to an Enum :)